sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -57,7 +60,7 @@ case class CreateDatabase(
       CatalogDatabase(
         databaseName,
         comment.getOrElse(""),
-        path.getOrElse(catalog.getDefaultDBPath(databaseName)),
+        catalog.createDatabasePath(databaseName, path),
```

Comment:
looking at it again. Seems we do not really need to pass in `path`, right? Also, how about we keep the name getDefulatDBPath because `createDatabasePath` implied file system operations (sorry....). But, it will be good to add comment to this method to explain its purpose.